packages/core-app-api/src/apis/implementations/StorageApi/WebStorage.ts

```diff
@@ -89,9 +89,19 @@ export class WebStorage implements StorageApi {
   observe$<T extends JsonValue>(
     key: string,
   ): Observable<StorageValueSnapshot<T>> {
+    this.addStorageEventListener();
     return this.observable.filter(({ key: messageKey }) => messageKey === key);
   }
 
+  private addStorageEventListener() {
+    window.addEventListener('storage', event => {
+      if (event.key?.includes(this.namespace)) {
```

Comment:
This check is probably a bit too liberal, would be best to check for ```namespace === event.key || namespace.startsWith(`${event.key}/`)``` and then looking at what that means for the replacement below (when there's an exact match with no slash)

Reply:
@freben Thanks, `startsWith` is a better check for this than `includes` .

> namespace.startsWith(`${event.key}/`)

I think that will never be `true` since `event.key` contains `namespace + key`. You mean `event.key.startsWith(namespace)`?

Reply:
Not sure what I meant :D Just in such a way that it works, and never gets false positives by finding matches higher up in the string instead of at the start